oshi-core/src/main/java/oshi/hardware/common/AbstractNetworkIF.java

```diff
@@ -170,6 +164,11 @@ public String getName() {
         return this.networkInterface.getName();
     }
 
+    @Override
+    public int getIndex() {
+        return this.networkInterface.getIndex();
```

Comment:
This works, but looking at `AbstractNetworkInterface` we already grab many of the other relevant values (`name`, `mtu`, etc.) when we construct the object and it makes sense to add a `private int index` variable, populate it in the constructor, and simply return that.